src/Modal/Modal.tsx

```diff
@@ -0,0 +1,108 @@
+import * as React from "react";
+import "./modal-style.scss";
+
+interface ModalProps {
+    toggle: boolean;
+    id?: string;
+    fullscreen?: boolean;
+    position?: "left" | "right";
+    className?: string;
+    disableBackdropDismiss?: boolean;
+    ariaLabel?: string;
+    ariaDescribedby?: string;
+    dismissModal?: () => void;
+    header?: React.ReactNode;
+    body?: React.ReactNode;
+    footer?: React.ReactNode;
+}
+
+interface ModalState {
+    open: boolean;
+    close: boolean;
+}
+
+export class Modal extends React.Component<ModalProps, ModalState> {
+    constructor(props: ModalProps) {
+        super(props);
+        this.state = {
+            open: false,
+            close: false
+        };
+    }
+
+    componentDidMount(): void {
+        if (this.props.toggle === true) {
+            this.setState({ open: true });
+        }
+    }
+
+    componentDidUpdate(prevProps: ModalProps): void {
+        if (this.props.toggle !== prevProps.toggle) {
+            this.setState({ open: !prevProps.toggle, close: prevProps.toggle });
+        }
+    }
+
+    /**
+     * set modal class names
+     * @returns string of concatenated class names
+     */
+    setModalClasses(): string {
+        const classes: string = this.classNames({
+            "show": this.state.open,
+            "fade": this.state.close,
+            "modal-aside": !!this.props.position,
+            "modal-aside-left": this.props.position === "left",
+            "modal-aside-right": this.props.position === "right",
+            "modal-fullscreen": this.props.fullscreen,
+            [this.props.className]: !!this.props.className
+        });
```

Comment:
I don't know why you chose this to be a two-step process? Declaring an object and then processing this object. You could do this instead:
```typescript
let classes: string = this.props.toggle ? "show" : "hide";
classes += !!this.props.position ? " modal-aside" : "";
classes += !!this.props.position === "left" ? " modal-aside-left" : " modal-aside-right";
classes += this.props.fullscreen ? " modal-fullscreen" : "";
classes += this.props.className ? ` ${this.props.className}` : "";
return classes;
```
I think you're just using `ngClass` logic from `Angular` here and adding a method to process it which is unnecessary.

Reply:
i believe its a cleaner solution than concatenating string and adding spaces to class names.